tokenizer/udic.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2016 ikawaha
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// 	You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tokenizer
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strings"
+
+	"github.com/ikawaha/kagome/internal/dic"
+)
+
+// UserDic represents a user dictionary.
+type UserDic struct {
+	dic *dic.UserDic
+}
+
+// NewUserDic build a user dictionary from a file.
+func NewUserDic(path string) (UserDic, error) {
+	f, err := os.Open(path)
+	if err != nil {
+		return UserDic{}, err
+	}
+	defer f.Close()
+
+	r, err := NewUserDicRecords(f)
+	if err != nil {
+		return UserDic{}, err
+	}
+	return r.NewUserDic()
+}
+
+type UserDicRecord struct {
+	Text   string   `json:"text"`
+	Tokens []string `json:"tokens"`
+	Yomi   []string `json:"yomi"`
+	Pos    string   `json:"pos"`
+}
+
+type UserDicRecords []UserDicRecord
+
+func (u UserDicRecords) Len() int           { return len(u) }
+func (u UserDicRecords) Swap(i, j int)      { u[i], u[j] = u[j], u[i] }
+func (u UserDicRecords) Less(i, j int) bool { return u[i].Text < u[j].Text }
+
+func NewUserDicRecords(r io.Reader) (UserDicRecords, error) {
```

Comment:
__[gometalinter]__
* [golint] exported function NewUserDicRecords should have comment or be unexported